### PR TITLE
[WebView] Add WebView Settings API

### PR DESCRIPTION
**Pull request by BasavarajPS**
Tizen Settings API

Signed-off-by: Basavaraj P S <basavaraj.ps@samsung.com>

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by BasavarajPS** [comment]
Submitted new patch with changes done

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 1]
```diff
@@ -98,5 +115,63 @@ public int DefaultFontSize
...
+
+        /// <summary>
+        /// Whether the autofill password form enabled.
+        /// </summary>
+        /// <since_tizen> 4 </since_tizen>
+        public bool AutoFillPasswordForm
+        {
+            set
+            {
+                Interop.ChromiumEwk.ewk_settings_autofill_password_form_enabled_set(_handle, value);
```
[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: @BasavarajPS If get API is not Public, how about change to method instead property?

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 2]
```diff
@@ -98,5 +115,75 @@ public int DefaultFontSize
...
+            {
+                Interop.ChromiumEwk.ewk_settings_scripts_can_open_windows_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the view port meta tag enabled.
+        /// </summary>
+        /// <since_tizen> 4 </since_tizen>
+        public bool ViewPortMetaTagEnabled
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: What is `ViewPortMetaTag`? 
Would you explain more details?

I want to more summary

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 3]
```diff
@@ -98,5 +115,75 @@ public int DefaultFontSize
...
+        public bool FormCandidateDataEnabled
+        {
+            set
+            {
+                Interop.ChromiumEwk.ewk_settings_form_candidate_data_enabled_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the form profile data enabled.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: What is `FormProfileData` ?

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 4]
```diff
@@ -98,5 +115,75 @@ public int DefaultFontSize
...
+        public bool AutoFillPasswordFormEnabled
+        {
+            set
+            {
+                Interop.ChromiumEwk.ewk_settings_autofill_password_form_enabled_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the form candidate data enabled.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: What is `FormCandidateData`?

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 2 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: view port meta tag is used to control the layout on the device.
https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 3 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: it is AutoFill form

Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: It is a just example to use enum class to control autofill option

``` c#
	[Flags]
	public enum AutoFillMode
	{
		None = 0,
		Forms = 1 << 1,
		Password = 1 << 2,
		Suggestion = 1 << 3,
	}

    void SetAutoFillMode(AutoFillMode mode) 
    {
    	Interop.ChromiumEwk.ewk_settings_form_profile_data_enabled_set(_handle, mode.HasFlag(AutoFillMode.Forms));
		Interop.ChromiumEwk.ewk_settings_autofill_password_form_enabled_set(_handle, mode.HasFlag(AutoFillMode.Password));
		Interop.ChromiumEwk.ewk_settings_form_candidate_data_enabled_set(_handle, mode.HasFlag(AutoFillMode.Suggestion));
    }

```

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 2 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 3 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 4 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: Candidate autofill form is for suggestion

**Comment by BasavarajPS** [comment]
Please check my latest patch

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 5]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+
+            set
+            {
+                Interop.ChromiumEwk.ewk_settings_scripts_can_open_windows_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the viewport meta element is enabled.
+        /// If true, Viewport Meta element is applied.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: Meta -> meta

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 6]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+            /// </summary>
+            Password = 1 << 2,
+
+            /// <summary>
+            /// autofill suggestion enabled.
+            /// </summary>
+            Suggestion = 1 << 3,
+        }
+
+	void setAutoFill(AutoFillMode mode)
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: Why is not public?

setAutoFill -> SetAutofillMode

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 7]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+                Interop.ChromiumEwk.ewk_settings_viewport_meta_tag_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the autofill enabled.
+        /// </summary>
+        /// <since_tizen> 4 </since_tizen>
+	[Flags]
+	public enum AutoFillMode
```
[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: AutoFillMode -> AutofillModes

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 8]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+            }
+        }
+
+        /// <summary>
+        /// Whether the viewport meta element is enabled.
+        /// If true, Viewport Meta element is applied.
+        /// If false, Viewport meta element is ignored.
+        /// </summary>
+        /// <since_tizen> 4 </since_tizen>
+        public bool ViewPortEnabled
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: ViewPortEnabled -> ViewportEnabled

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 9]
```diff
@@ -31,6 +31,23 @@ internal Settings(IntPtr handle)
             _handle = handle;
         }
 
+        /// <summary>
+        /// Whether the Auto Fitting can be fit to width.
+        /// </summary>
+        /// <since_tizen> 4 </since_tizen>
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: since_tizen should be 6 for master branch.
Plz refer to Branches section of README.md.
Plus, you have to do ACR process as well, in order to add new API.

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 10]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+                Interop.ChromiumEwk.ewk_settings_viewport_meta_tag_set(_handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Whether the autofill enabled.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+	[Flags]
+	public enum AutofillModes
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: do not use inner enum

please extract to outside of this class

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 11]
```diff
@@ -98,5 +115,76 @@ public int DefaultFontSize
...
+            /// </summary>
+            Password = 1 << 2,
+
+            /// <summary>
+            /// autofill suggestion enabled.
+            /// </summary>
+            Suggestion = 1 << 3,
+        }
+
+	void SetAutofillMode(AutofillModes mode)
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/Settings.cs]: why not public?

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 10 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: done

File: src/Tizen.WebView/Tizen.WebView/Settings.cs (hunk 11 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: done

File: src/Tizen.WebView/Tizen.WebView/Settings.cs [hunk 12]
```diff
@@ -98,5 +143,48 @@ public int DefaultFontSize
...
+                return Interop.ChromiumEwk.ewk_settings_viewport_meta_tag_get(_handle);
+            }
+
+            set
+            {
+                Interop.ChromiumEwk.ewk_settings_viewport_meta_tag_set(_handle, value);
+            }
+        }
+
+	public void SetAutofillMode(AutofillModes mode)
```
[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: please add descriptions

[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/Settings.cs]: fix indent

**Comment by myroot** [comment]
Please preceding ACR process

**Comment by BasavarajPS** [comment]
new set of ewk setting APis should be done
This patch is not needed

**Pull request closed without merging**